### PR TITLE
PR: Use only tab to select completions

### LINKED ISSUE #10134: On the fly code completion interferes with normal typing
### What steps will reproduce the problem?

This may be intended behavior, however, it didn't happen with 3.X. 

With code completion and on the fly code completion turned on (note these are on different panes in preferences, which makes discovery hard), try to type the following function definition in a new editor window.

    def f(x):
        return x

The code completion pops up a window with xrange as the completion of the `return x`. If you are typing quickly (as I do) it will auto-complete x as xrange, yielding

    def f(x):
        return xrange

I think 3.x had a way to control whether ENTER auto completes or not, I don't think I see that in 4 (this issue reporter window is modal, so I can't check while I am typing this)

Anyway, this may be the intended/desired behavior, but I find it hard to imagine most will find it useful. One may counter that x is a bad variable name, but I only really do math in Python, so x is very common usage. 

## Versions

* Spyder version: 4.0.0b4 
* Python version: 3.7.3
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 7

### Dependencies

```
pygments >=2.0    :  2.4.2 (OK)
qdarkstyle >=2.6.4:  2.7 (OK)
sphinx >=0.6.6    :  2.1.2 (OK)
pyls >=0.27.0     :  0.28.2 (OK)
nbconvert >=4.0   :  5.5.0 (OK)
pandas >=0.13.1   :  0.25.1 (OK)
numpy >=1.7       :  1.16.4 (OK)
sympy >=0.7.3     :  1.4 (OK)
cython >=0.21     :  0.29.13 (OK)
qtconsole >=4.5.2 :  4.5.4 (OK)
IPython >=4.0     :  7.8.0 (OK)
matplotlib >=2.0.0:  3.1.1 (OK)
pylint >=0.25     :  2.3.1 (OK)
```

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #10134

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by ccordoba12** [comment]
@goanpeca, I think you need to fix our completion tests to use Tab instead of Enter.

File: spyder/plugins/editor/widgets/completion.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 from spyder.widgets.helperwidgets import HTMLDelegate
 
 
-COMPLETION_ITEM_TEMPLATE = """
+COMPLETION_ITEM_TEMPLATE = u"""
```
Comment (goanpeca) [inline spyder/plugins/editor/widgets/completion.py]: Either we add this `u` or `unicode_literals` import on top.

Any preference @ccordoba12 ? (this was failing for me on python 2)

File: spyder/plugins/editor/widgets/completion.py [hunk 2]
```diff
@@ -277,14 +277,30 @@ def hide(self):
         QListWidget.hide(self)
         QToolTip.hideText()
 
+    def event(self, event):
+        """Override Qt method to process tab keypress."""
```
Comment (goanpeca) [inline spyder/plugins/editor/widgets/completion.py]: The `Tab` key is not actually propagated to he keypress so I had to filter it here

**Comment by goanpeca** [comment]
Not sure what the failure on Mac is @ccordoba12

**Comment by CAM-Gerlach** [comment]
@goanpeca Could you considering making this a configurable option, like it was in Spyder 3 (which had return as default) and a number of other IDEs? Otherwise, it will be a regression to hardcode it, whereas (particularly if the other suggestions in #10134 are implemented) return for completion offers substantial ergonomic, speed and consistency benefits over tab for many users and is a strong per-user personal preference. Thanks.

File: spyder/plugins/editor/widgets/completion.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/plugins/editor/widgets/completion.py]: Using `u` is fine. `unicode_literals` can introduce undesired side effects, that's why we don't use it.

**Comment by ccordoba12** [comment]
Let's follow the approach of PR #10262 instead.

**Pull request closed without merging**